### ISSUE TITLE
Could not find keyboard map for locale 0X00000000040E040E!

### ISSUE THREAD

**Issue by mEtzG3r** [kwhat/jnativehook#263]
Hi! I cannot find what can be the problem. The logger only says this (Could not find keyboard map for locale 0X00000000040E040E!) when I type some special letters, like É, Á, Ő etc. Can you help? If it's not solvable by choosing locale, is it possible to get the original keycode to map it? How to do it? Thanks in advance!

**Comment by kwhat** [comment]
Can you test the latest version

https://oss.sonatype.org/content/repositories/snapshots/com/1stleg/jnativehook/2.1.SNAPSHOT/jnativehook-2.1.20190725.024655-45.jar

**Comment by mEtzG3r** [comment]
I cannot download that jar file, Firefox, Chrome and Opera says that it's empty. Also, there was a jar file on Github too which is gone now.

**Comment by kwhat** [comment]
Sorry, looks like I added the wrong link.  Try https://oss.sonatype.org/content/groups/public/com/1stleg/jnativehook/2.1.SNAPSHOT/jnativehook-2.1.20190725.024655-45.jar

**Comment by mEtzG3r** [comment]
I removed the original 2.1.0 jar with Netbeans and added this. The problem is that there was working mouse and keyboard reaction but now nothing. The logger can list the actions, but the software isn't recognizing them anymore. How come? What am I doing wrong?

**Comment by kwhat** [comment]
Which platform?

**Comment by mEtzG3r** [comment]
x64 with Windows 10. I created a completely new example to test the provided 2.1.2 jar, but still no luck, nothing happens when I press the keyboard buttons. If I turn on the logger, it reacts to the keyboard, but the events never fire up:

package jnativetest;
        
import java.util.logging.Level;
import java.util.logging.Logger;
import org.jnativehook.GlobalScreen;
import org.jnativehook.NativeHookException;
import org.jnativehook.keyboard.NativeKeyEvent;
import org.jnativehook.keyboard.NativeKeyListener;

public class Jnativetest implements NativeKeyListener {
	public void nativeKeyPressed(NativeKeyEvent e) {
		System.out.println("Key Pressed: " + NativeKeyEvent.getKeyText(e.getKeyCode()));

		if (e.getKeyCode() == NativeKeyEvent.VC_ESCAPE) {
			if (e.getKeyCode() == NativeKeyEvent.VC_ESCAPE) {
				try {
					GlobalScreen.unregisterNativeHook();
				} catch (NativeHookException ex) {
					ex.printStackTrace();
				}
			}
		}
	}

	public void nativeKeyReleased(NativeKeyEvent e) {
		System.out.println("Key Released: " + NativeKeyEvent.getKeyText(e.getKeyCode()));
	}

	public void nativeKeyTyped(NativeKeyEvent e) {
		System.out.println("Key Typed: " + e.getKeyText(e.getKeyCode()));
	}

	public static void main(String[] args) {
                Logger logger = Logger.getLogger(GlobalScreen.class.getPackage().getName());
                logger.setLevel(Level.OFF);
            
		try {
			GlobalScreen.registerNativeHook();
                        System.err.println("Nativehook registered.");
		}
		catch (NativeHookException ex) {
			System.err.println("There was a problem registering the native hook.");
			System.err.println(ex.getMessage());

			System.exit(1);
		}

		GlobalScreen.addNativeKeyListener(new Jnativetest());
                System.err.println("Nativehook listener online.");
                System.err.println("Waiting for input...");
	}
}

**Comment by durandjerms** [comment]
Hi,
Same problem as mEtzG3r with this jar.
Windows 10 x64.

events do not fire and on the logger i have this when i press a key : 
_1 août 2019 10:29:42 org.jnativehook.GlobalScreen$NativeHookThread enable
INFO: process_key_pressed [252]: Key 0 typed. (a)

1 août 2019 10:29:42 org.jnativehook.GlobalScreen$NativeHookThread enable
INFO: process_key_released [286]: Key 0X1E released. (0X41)_

**Comment by LukaszWiktor** [comment]
Same issue here, just different locale ids:
```
Sep 14, 2019 8:17:30 AM org.jnativehook.GlobalScreen$NativeHookThread enable
SEVERE: refresh_locale_list [628]: Could not find keyboard map for locale 0X0000000004090409!

Sep 14, 2019 8:17:30 AM org.jnativehook.GlobalScreen$NativeHookThread enable
SEVERE: refresh_locale_list [628]: Could not find keyboard map for locale 0X0000000004150809!

Sep 14, 2019 8:17:30 AM org.jnativehook.GlobalScreen$NativeHookThread enable
SEVERE: refresh_locale_list [628]: Could not find keyboard map for locale 0X0000000008090809!
```

**Comment by kwhat** [comment]
Please test [jnativehook-2.2-20210107.030301-37.jar](https://oss.sonatype.org/content/repositories/snapshots/com/github/kwhat/jnativehook/2.2-SNAPSHOT/jnativehook-2.2-20210107.030301-37.jar)  I have done some refactoring to the function that is probably responsible for this issue but because I cannot reproduce the problem, I need one of you to test it and let me know if it is working.  I still don't know exactly why this was an issue nor why I cannot reproduce it.

**Comment by kwhat** [comment]
I suspect this maybe fixed in 2.2.0, please test and re-open if this is still an issue.
